src/FOM/UserBundle/Controller/UserController.php

```diff
@@ -30,21 +33,71 @@ class UserController extends Controller
     public function indexAction()
     {
         $securityContext = $this->get('security.context');
-        $oid             = new ObjectIdentity('class', 'FOM\UserBundle\Entity\User');
+        $oid = new ObjectIdentity('class', 'FOM\UserBundle\Entity\User');
+        return array(
+            'users'             => array(''),
+            'create_permission' => $securityContext->isGranted('CREATE', $oid));
+    }
 
-        $query         = $this->getDoctrine()->getManager()->createQuery('SELECT r FROM FOMUserBundle:User r');
-        $users         = $query->getResult();
-        $allowed_users = array();
-        // ACL access check
-        foreach ($users as $index => $user) {
+    /**
+     * Renders user list.
+     *
+     * @ManagerRoute("/user/search")
+     * @Method({ "POST","GET" })
+     */
+    public function searchAction()
+    {
+        $securityContext = $this->get('security.context');
+        $oid = new ObjectIdentity('class', 'FOM\UserBundle\Entity\User');
+        $request = Request::createFromGlobals();
+        $start = $request->query->get('start');
+        $length = $request->query->get('length');
+        $search = $request->query->get('search');
+        $dql = "SELECT r FROM FOMUserBundle:User r LEFT JOIN r.groups g WHERE
```

Comment:
I suggest you open an issue for that (enhancement - replace ObjectIdentity with FOMIdentitiesProvider).
My intention wasn't a complete refactoring of the code...